### PR TITLE
QPPC-4141 Add script to compare CSV Single source

### PR DESCRIPTION
**Pull request by aokasb**
#### Motivation for change

We need to validate the single source file and what has changed since past year in that. Thus we need a method to quickly compare and draw the attention to the changes and verify them in PDF. Verification generates a Report in Markdown like style.  

#### What is being changed

A script is added to create markdown files which can be reviewed by experienced analyst/ developer.   A couple of examples and summary is also enclosed. The script can also take measure_id as input in order to only find differences in that measure

#### Release checklist:
* [X] Verified working locally

Quick verification  ( See attached sample output ) 
``` 
diff <(grep "^14," ../../data/2021_Claims_SingleSource_v1.3.csv | grep ENCOUNTER_CODE|sed 's/"//g' |sort) <(grep "^14," ../../data/2020_Claims_SingleSource_v1.4.csv | grep ENCOUNTER_CODE|sort|sed 's/"//g')
4a5
> 14,ENCOUNTER_CODE,C4,99201,≠GQ, GT, 95,≠02,≥50,M, F
```

##### Associated JIRA tickets:
* https://jira.cms.gov/browse/QPPC-4141

### CODE REVIEW

File: claims-related/tests/single_source_csv/csv_report/Measure014.md
Comment (chad-semanticbits) [inline claims-related/tests/single_source_csv/csv_report/Measure014.md:13]: This doesn't actually display as a table in markdown; it's just one long line.  Getting an actual markdown table might be more trouble than it's worth, but it would be nice to do a list of key: value pairs on new lines.

VERSION: BASE
Measure_ID: 14
...

File: claims-related/tests/single_source_csv/csv_report/Measure014.md
Comment (chad-semanticbits) [inline claims-related/tests/single_source_csv/csv_report/Measure014.md:2]: These lines show up as one long line in a markdown viewer.  I think you need to add an extra newline.

File: claims-related/tests/single_source_csv/single_source_csv_comp.py
Comment (chad-semanticbits) [inline claims-related/tests/single_source_csv/single_source_csv_comp.py:112]: I don't know if you want to keep these commented out lines

File: claims-related/tests/single_source_csv/csv_report/Measure014.md
Comment (chad-semanticbits) [reply claims-related/tests/single_source_csv/csv_report/Measure014.md:2]: Actually, it might just be an issue with my VS Code extension

**Review by chad-semanticbits — DISMISSED** [review]

File: claims-related/tests/single_source_csv/single_source_csv_comp.py
Comment (panha-yorn) [inline claims-related/tests/single_source_csv/single_source_csv_comp.py:17]: we can remove this comment

Comment (aokasb) [reply claims-related/tests/single_source_csv/single_source_csv_comp.py:17]: @panha-yorn  If we execute this locally we can use that comment assuming that we have the to_markdown function. These are off-hand scripts that are executed by user and not necessarily need to use the recommended mechanism to execute. The idea was to use to_markdown which is *much nicer* but I am not committing it because of compatibility issue.

File: claims-related/tests/single_source_csv/single_source_csv_comp.py
Comment (aokasb) [reply claims-related/tests/single_source_csv/single_source_csv_comp.py]: This is Done

File: claims-related/tests/single_source_csv/csv_report/Measure014.md
Comment (aokasb) [reply claims-related/tests/single_source_csv/csv_report/Measure014.md:2]: I have made sure that these are long lines so that they can be viewed as tabular files ( Ideally wanted them as markdown )

File: claims-related/tests/single_source_csv/single_source_csv_comp.py
Comment (aokasb) [reply claims-related/tests/single_source_csv/single_source_csv_comp.py:18]: Done added a comment specifying usage

**Review by panha-yorn — APPROVED** [review]
LGTM. Please get approvals form a DS and an external folk who also works on this repo before merging it.

**Review by brayellison — APPROVED** [review]

**Comment by aokasb** [comment]
@KyleApfel  Can we merge this PR ? It adds an offline script which can compare two single source files finding the differences between them.

**Review by KyleApfel — APPROVED** [review]
Great stuff!

**Pull request merged**
